drivers/hmis/app/graphql/types/hmis_schema/ce_referral.rb

```diff
@@ -10,43 +10,70 @@ module Types
   class HmisSchema::CeReferral < Types::BaseObject
     # object is a Hmis::Ce::Referral
 
-    field :id, ID, null: false
-    field :opportunity, HmisSchema::CeOpportunity, null: false
-    field :steps, [HmisSchema::CeReferralStep], null: false
-    field :status, HmisSchema::Enums::CeReferralStatus, null: false
-    field :custom_status, HmisSchema::CeCustomReferralStatus, null: true
-    field :client_id, ID, null: false
-    field :client_name, String, null: false, description: 'The name of the referred client. Always available, even without full client record access.'
-    field :client_age, Integer, null: true, description: 'The age of the referred client. Always available, even without full client record access.'
-    field :client, Types::HmisSchema::Client, null: true, description: 'The full client record, if the user has permission to view it.'
-    field :created_at, GraphQL::Types::ISO8601DateTime, null: false
+    # Similar to Enrollment, this schema class overrides the `field` method
+    # and adds a `summary_field` method, to distinguish between fields that can be resolved
+    # by those who can_view? vs. just can_view_summary?
+    # If user lacks sufficient access, the field will be resolved as null.
+    def self.field(name, type = nil, **kwargs)
+      # See Types::BaseField `authorized?` function.
+      can_view_lambda = lambda do |current_user, object|
+        current_user.policy_for(object, policy_type: :ce_referral).can_view?
+      end
+
+      super(name, type, authorize_with: can_view_lambda, **kwargs)
```

Comment:
Seems like this should conditionally set authorize_with?
```suggestion
      # See Types::BaseField `authorized?` function.
      unless kwargs.key?(:authorize_with)
        kwargs[:authorize_with] = lambda do |current_user, object|
          current_user.policy_for(object, policy_type: :ce_referral).can_view?
        end
      end
        
      super(name, type, **kwargs)
```

Reply:
Makes sense! I think this mirrors a similar unsurfaced bug in the Enrollment schema object? https://github.com/greenriver/hmis-warehouse/blob/db323483a2d0dfe0d439a0e8e82e75138d40ad0e/drivers/hmis/app/graphql/types/hmis_schema/enrollment.rb#L61 (Unsurfaced because we don't currently override the `permissions` kwarg on any fields in that file.) But I think I would prefer to leave that unchanged for now to avoid qa scope creep.